include/rapidcheck/state/Commands.hpp

```diff
@@ -1,9 +1,109 @@
 #pragma once
 
 #include "rapidcheck/detail/Results.h"
+#include "rapidcheck/state/gen/ParallelCommands.hpp"
+#include "rapidcheck/Assertions.h"
+#include <thread>
+#include <mutex>
+#include "rapidcheck/detail/Barrier.h"
 
 namespace rc {
 namespace state {
+namespace detail {
+
+
+template <typename Model, typename Cmd>
```

Comment:
`Model` shouldn't be needed as a type argument here since that's a function of the command `Cmd`.